### PR TITLE
Added setup.sh

### PR DESCRIPTION
**Pull request by Timothyw0**

### CODE REVIEW

File: bin/setup.sh [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+./configure
+make
+sudo make install
+cd ..
+
+# Set up PostGIS
+echo "Setting up PostGIS"
+echo
+echo
+sudo su - postgres -c psql django_test < psqlCommands.sql
```
Comment (SillyInventor) [inline bin/setup.sh]: Errors with no such file or directory (psqlCommands.sql)

File: bin/setup.sh [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+echo
+wget http://download.osgeo.org/proj/proj-4.8.0.tar.gz
+wget http://download.osgeo.org/proj/proj-datumgrid-1.5.tar.gz
+tar xzf proj-4.8.0.tar.gz
+cd proj-4.8.0/nad
+tar xzf ../../proj-datumgrid-1.5.tar.gz
+cd ..
+./configure
+make
+sudo make install
```
Comment (SillyInventor) [inline bin/setup.sh]: Same as above.

File: bin/setup.sh [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+# Install GEOS
+echo "Installing GEOS"
+echo
+echo
+wget http://download.osgeo.org/geos/geos-3.4.2.tar.bz2
+tar xjf geos-3.4.2.tar.bz2
+cd geos-3.4.2
+./configure
+make
+sudo make install
```
Comment (SillyInventor) [inline bin/setup.sh]: Errors no C compiler found.

File: bin/setup.sh [hunk 4]
```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+
+# Essentials
+echo "Getting essentials"
+echo
+echo
+sudo apt-get install python-dev build-essential
```
Comment (SillyInventor) [inline bin/setup.sh]: Must have option -y, otherwise aborts.

File: bin/setup.sh [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+# Pip and Virtualenv
+echo "Getting pip, virtualenv, and pip libraries"
+echo
+echo
+sudo apt-get install python-pip
```
Comment (SillyInventor) [inline bin/setup.sh]: Same as the above.

**Comment by Timothyw0** [comment]
Errors should be fixed now

**Comment by SillyInventor** [comment]
Still erring. Make sure you're running vagrant destroy and vagrant up and look at the log.

**Comment by Timothyw0** [comment]
Should be good now

File: bin/setup.sh [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+./configure
+make
+sudo make install
+cd ..
+
+# Set up PostGIS
+echo "Setting up PostGIS"
+echo
+echo
+cd ~/
```
Comment (SillyInventor) [inline bin/setup.sh]: You must first create the database

File: bin/setup.sh [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+echo "CREATE USER django_user WITH PASSWORD 'dj4ng0_t3st';CREATE EXTENSION adminpack;CREATE EXTENSION postgis;CREATE EXTENSION postgis_topology;\q" >> psqlCommands.sql
```
Comment (SillyInventor) [inline bin/setup.sh]: You can just call: sudo -u postgres psql django_test-c "CREATE EXTENSION adminpack; CREATE EXTENSION postgis; CREATE EXTENSION postgis_topology;"

File: bin/setup.sh [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+echo "Getting pip, virtualenv, and pip libraries"
+echo
+echo
+sudo apt-get -y install python-pip
+sudo pip install virtualenv
+sudo apt-get -y install libcurl4-openssl-dev
+sudo apt-get -y install libpq-dev
+. activate
+pip install https://github.com/djangonauts/django-rest-framework-gis/tarball/master
+
```
Comment (SillyInventor) [inline bin/setup.sh]: make sure you install the libraries from libraries.txt (should be changed to requirements.txt)

File: Vagrantfile [hunk 9]
```diff
@@ -24,12 +24,14 @@ Vagrant.configure("2") do |config|
   # accessing "localhost:8080" will access port 80 on the guest machine.
   # NOTE: This will enable public access to the opened port
   # config.vm.network "forwarded_port", guest: 80, host: 8080
+  config.vm.network "forwarded_port", guest: 80, host: 8080
```
Comment (SillyInventor) [inline Vagrantfile]: Don't use 80, that's a special port. Use something >8000.

File: bin/setup.sh (hunk 6 above)
Comment (Timothyw0) [reply bin/setup.sh]: The database is created in line 35

File: bin/setup.sh (hunk 7 above)
Comment (Timothyw0) [reply bin/setup.sh]: This is very helpful, thank you

File: bin/setup.sh (hunk 6 above)
Comment (SillyInventor) [reply bin/setup.sh]: Hmm, well, I had to make a new database when I used vagrant up. It was complaining that no such database existed.

Comment (Timothyw0) [reply bin/setup.sh]: I'm not sure, I'll look into it

File: bin/setup.sh (hunk 8 above)
Comment (Timothyw0) [reply bin/setup.sh]: I'm not sure how to do this, I have to call pip install -r requirements.txt but how can I get that text file onto the vagrant server?

Comment (SillyInventor) [reply bin/setup.sh]: sudo pip install --upgrade -r /vagrant/libraries.txt

File: bin/setup.sh [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+sudo apt-get -y install libxml2-dev
+sudo apt-get -y install libgdal1-dev
+sudo apt-get -y install postgis*
+wget http://download.osgeo.org/postgis/source/postgis-2.4.3.tar.gz
+tar xvzf postgis-2.4.3.tar.gz
+cd postgis-2.4.3
+./configure
+make
+make install
+sudo su - postgres -c createdb django_test exit
```
Comment (SillyInventor) [inline bin/setup.sh]: Someone forgot quotes. And you don't need exit.

File: bin/setup.sh [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+./configure
+make
+sudo make install
+cd ..
+
+# Set up PostGIS
+echo "Setting up PostGIS"
+echo
+echo
+sudo -u postgres psql django_test-c "CREATE USER django_user WITH PASSWORD 'dj4ng0_t3st';CREATE EXTENSION adminpack; CREATE EXTENSION postgis; CREATE EXTENSION postgis_topology;"
```
Comment (SillyInventor) [inline bin/setup.sh]: You forgot a space between django_test and -c.

**Comment by SillyInventor** [comment]
YAY! Good job.

**Pull request merged**